stories/10-For Developers/Token Implementation Notes.stories.mdx

```diff
@@ -0,0 +1,229 @@
+```
+
+```
+// DON'T DO THIS
+.my-button {
+  @include typeface (body, 6);
+  line-height: 100%; // customized line height for buttons.
+}
+
+.my-button.mod-small {
+  @include typeface (body, 4); // overwrites the customized line height. oops!
+}
+```
+
+## Spacing
+
+Radial includes three predefined spacing scales, Compact, Default, and Loose. To choose the spacing scale for a theme, the theme author can change the variables `--spacing-base` and `spacing-increment` to use a pair of these predefined variables.
+
+Setting a theme to use compact spacing:
+```
+  --spacing-base:      var(--spacing-base-compact);
+  --spacing-increment: var(--spacing-increment-compact);
+```
+
+Setting these levers configures the values for a range of spacing tokens, numbered `--space-1` to `--space-11`.
+
+When creating components, developers and designers should use these tokens consistely according to the spacing guidelines.
+
+## Colors
+
+### Color Format
+
+One nice feature of SASS is its built in color functions. You can write `rgba(#ff000, 0.8)` and it will become `rgba(255, 0, 0, 0.8)`. Or more usefully you might write something like `rgba($theme-red, $hover-opacity)`.
+
+Unfortunately things aren't so simple in pure CSS.
+
+The problem:
+We want to be able to store colors in a way where:
+
+1. We can apply opacity to the colors.
+2. The variable names are preserved as CSS variables even after compilation.
+
+Obstacles:
+- SASS variables wont work because the variable names get compiled out.
+- Supported CSS color formats won't let use rgb colors:
+- Using the 8 digit hex format (#RRGGBBAA) almost works, but we can't concatenate css vars without a space in-between. `--my-green: '#229922'; --my-opacity: 'cc'; color: var(--my-green)var(--my-opacity);` doesn't work. nor does `--my-green: '#229922'; color: var(--my-green)cc;`.
+-  `--my-green: '#229922'; rgba(#229922, 0.8)` also won't work in plain CSS.
+
+We ended up storing colors as a comma separated set of rgb values: e.g. `--my-green: "34,153,34";`
+This lets us do `color: rgba(var(--my-green), 0.8)` or even `color: rgba(var(--my-green), (--opacity-token))`
+
+Since the design team uses hex colors in their designs, we include a sass function that will convert hex colors to these rgb values, to make theme files easier to parse.
+
+`--my-green: hex2rgb("229922");` becomes `--my-green: "34,153,34"` in the compiled css.
+
+I didn't find this until after we sorted this all out, but the first part of this blog post covers this issue and came to basically this exact same solution: https://blog.jim-nielsen.com/2019/generating-shades-of-color-using-css-variables/
+
+### The base palette
+
+Because color tokens will be need to be customized for each theme, and also documented in several places we don't want to go overboard with the list of color tokens. The base set should be colors that are use widely in multiple places across our sites and provide a useful default pallete that we can use to build our components.
```

Comment:
```suggestion
Because color tokens will need to be customized for each theme and documented in several places, we don't want to go overboard with the list of color tokens. The base set should include colors that are used widely in multiple places across our sites and provide a useful default palette that we can use to build our components.
```